ddtrace/internal/gitmetadata.py

```diff
@@ -0,0 +1,128 @@
+import typing
+
+from envier import Env
+
+from ddtrace.internal.utils import formats
+
+
+_GITMETADATA_TAGS = None  # type: typing.Optional[typing.Dict[str, str]]
+
+TAG_REPOSITORY_URL = "git.repository_url"
+TAG_COMMIT_SHA = "git.commit.sha"
+
+TRACE_TAG_REPOSITORY_URL = "_dd.git.repository_url"
+TRACE_TAG_COMMIT_SHA = "_dd.git.commit.sha"
+
+
+class GitMetadataConfig(Env):
+    __prefix__ = "dd"
+
+    # DD_TRACE_GIT_METADATA_ENABLED
+    enabled = Env.var(bool, "trace_git_metadata_enabled", default=True)
+
+    # DD_GIT_REPOSITORY_URL
+    repository_url = Env.var(str, "git.repository_url", default="")
+
+    # DD_GIT_COMMIT_SHA
+    commit_sha = Env.var(str, "git.commit.sha", default="")
+
+    # DD_MAIN_PACKAGE
+    main_package = Env.var(str, "main_package", default="")
+
+    # DD_TAGS
+    tags = Env.var(str, "tags", default="")
+
+
+def _get_tags_from_env(config):
+    # type: (GitMetadataConfig) -> typing.Optional[typing.Dict[str, str]]
+    """
+    Get git metadata from environment variables
+    """
+    repository_url = config.repository_url
+    commit_sha = config.commit_sha
+
+    if not (repository_url and commit_sha):
+        tags = formats.parse_tags_str(config.tags)
+        repository_url = tags.get(TAG_REPOSITORY_URL)
+        commit_sha = tags.get(TAG_COMMIT_SHA)
+
+    if not (repository_url and commit_sha):
+        return None
```

Comment:
The original idea was to only report complete data, but if you think having partial data would also be useful, let's add

Reply:
Generally speaking I think the tracer user would expect each env var to work independently. Otherwise, as a tracer user. I'd expect a single env var containing both info.